apis/tempo/v1alpha1/tempostack_types.go

```diff
@@ -623,14 +623,23 @@ type QueryLimit struct {
 	// +kubebuilder:validation:Optional
 	// +operator-sdk:csv:customresourcedefinitions:type=spec,xDescriptors="urn:alm:descriptor:com.tectonic.ui:number",displayName="Max Tags per User"
 	MaxBytesPerTagValues *int `json:"maxBytesPerTagValues,omitempty"`
-	// MaxSearchBytesPerTrace defines the maximum size of search data for a single
+
+	// DEPRECATED. MaxSearchBytesPerTrace defines the maximum size of search data for a single
```

Comment:
Can we show a warning or error to the user when the option is used?

Reply:
It would be good, but where?

* operator logs during reconciling
* operator logs during an upgrade
* a new status condition
* restrict it in the validating webhook (that'd break the automated upgrade, but we can set it to 0 during the upgrade)

Any better ideas? I'm not sure what's a good place for it, except for the changelog/release notes.

Reply:
>operator logs during reconciling
> operator logs during an upgrade

I would avoid logs, usually users don't check them and even might not have permissions to do so.

I was thinking about validating webhook

Reply:
Good idea, I've added validation to the webhook and added a upgrade procedure to remove this field if it is set.